src/docs/product/issues/issue-details/performance-issues/regex-main-thread/index.mdx

```diff
@@ -0,0 +1,65 @@
+---
+title: "Regex on Main Thread"
+sidebar_order: 30
+redirect_from:
+  - /product/issues/performance-issues/regex-decoding-main-thread/
+description: "Learn more about Regex on Main Thread issues and how to diagnose and fix them."
+---
+
+The main, (or UI) thread in a mobile app is responsible for handling all user interaction and needs to be able to respond to gestures and taps in real time. If a long-running operation blocks the main thread, the app becomes unresponsive, impacting the quality of the user experience.
+
+Evaluating matches between strings and regular expressions can be long-running operations that may impact app responsiveness. If a long-running regular expression operation is detected on the main thread that can be offloaded to the background, Sentry will flag them it as an issue.
+
+## Detection Criteria
+
+Once [Profiling](/product/profiling/) has been enabled, Sentry will be able to detect Regex on Main Thread issues by looking for functions related to regular expressions that have been running on the main thread for at least 40ms. This threshold has been chosen so that we can see the function appear in a sufficient number of samples collected by the profile, (~10ms per sample x 4 = ~40ms total), before we detect the issue.
```

Comment:
```suggestion
[Profiling](/product/profiling/) needs to be enabled for Sentry to detect Regex on Main Thread issues. Sentry looks for functions related to regular expressions that have been running on the main thread for at least 40ms. This threshold was chosen so that the function appears in a sufficient number of samples collected by the profile (~10ms per sample x 4 = ~40ms total) before it gets detected as an issue.
```